### ISSUE TITLE
Seeing a large number of ghost errors when using cfn_lint

### ISSUE THREAD

**Issue by V4G4X** [mfussenegger/nvim-lint#808]
I see a large number of "ghost" diagnostics errors, like this:

<img width="1111" alt="Image" src="https://github.com/user-attachments/assets/af142087-1b91-4cff-b029-dbc60b100eb2" />

Picker shows 192 "ghost" errors and 2 "ghost" warnings.
Calling the linter directly does not show that many:
```sh
❯ cfn-lint src/handlers/espocpp/espnetworkocpp.yml
E3510 'arn:*:logs:*:*:*' does not match '^(arn:aws[A-Za-z\\-]*?:[^:]+:[^:]*(:(?:\\d{12}|\\*|aws)?:.+|)|\\*)$'
src/handlers/espocpp/espnetworkocpp.yml:394:17

W1032 {'Fn::Join': ['', ['arn:*:iot:*:', {'Ref': 'AWS::AccountId'}, ':topic/$aws/rules/esp_simple_ts_ingest/node/*']]} does not match '^(arn:aws[A-Za-z\\-]*?:[^:]+:[^:]*(:(?:\\d{12}|\\*|aws)?:.+|)|\\*)$' when 'Fn::Join' is resolved
src/handlers/espocpp/espnetworkocpp.yml:422:17

E3510 'arn:*:logs:*:*:*' does not match '^(arn:aws[A-Za-z\\-]*?:[^:]+:[^:]*(:(?:\\d{12}|\\*|aws)?:.+|)|\\*)$'
src/handlers/espocpp/espnetworkocpp.yml:447:17
```

I am facing this issue only with `cfn-lint`, and I don't know how to fix it.
I have a workaround that toggles virtual lines, but I'd prefer a more permanent fix that doesn't show errors where there are none.
